### PR TITLE
Add http port to EC2 security group

### PR DESCRIPTION
**Pull request by adamjohnson01**
##  Description
The EC2 provisioner was only allowing access to the control port of the exit node which meant that it would not accept client connections.

Related to #15 

## How Has This Been Tested?
```
Using provider: ec2
Requesting host: nice-moser5 in eu-west-1, from ec2
Host: i-0819bcca26592f705, status: creating
[1/500] Host: i-0819bcca26592f705, status: creating
[2/500] Host: i-0819bcca26592f705, status: creating
[3/500] Host: i-0819bcca26592f705, status: creating
[4/500] Host: i-0819bcca26592f705, status: initialising
...
[78/500] Host: i-0819bcca26592f705, status: active
Inlets OSS exit-node summary:
  IP: 34.253.222.172
  Auth-token: nPB42iNFJJXoWpR7sx6YWll5oYWspgViJ7wQ3QwetWkqX6RY0PvwvjKl2mDSSXMU

Command:
  export UPSTREAM=http://127.0.0.1:8000
  inlets client --remote "ws://34.253.222.172:8080" \
	--token "nPB42iNFJJXoWpR7sx6YWll5oYWspgViJ7wQ3QwetWkqX6RY0PvwvjKl2mDSSXMU" \
	--upstream $UPSTREAM

To Delete:
  inletsctl delete --provider ec2 --id "i-0819bcca26592f705"
```
```
 curl http://34.253.222.172 -v
*   Trying 34.253.222.172...
* TCP_NODELAY set
* Connected to 34.253.222.172 (34.253.222.172) port 80 (#0)
> GET / HTTP/1.1
> Host: 34.253.222.172
> User-Agent: curl/7.64.1
> Accept: */*
>
< HTTP/1.1 503 Service Unavailable
< Date: Fri, 10 Jan 2020 12:46:18 GMT
< Content-Length: 0
<
* Connection #0 to host 34.253.222.172 left intact
* Closing connection 0
```

Open all ports for Pro
```
Using provider: ec2
Requesting host: nifty-archimedes1 in eu-west-1, from ec2
Host: i-006960921893b7b48, status: creating
[1/500] Host: i-006960921893b7b48, status: creating
[2/500] Host: i-006960921893b7b48, status: creating
[3/500] Host: i-006960921893b7b48, status: initialising
...
[87/500] Host: i-006960921893b7b48, status: active
inlets-pro exit-node summary:
  IP: 18.203.111.129
  Auth-token: QpEpHpKdzjXXDPTG8AJUd945EipP883znsmJoTz6huiuoeKXEHrtW7XrLEOFhED6

Command:
  export TCP_PORTS="8000"
  export LICENSE=""
  inlets-pro client --connect "wss://18.203.111.129:8123/connect" \
	--token "QpEpHpKdzjXXDPTG8AJUd945EipP883znsmJoTz6huiuoeKXEHrtW7XrLEOFhED6" \
	--license "$LICENSE" \
	--tcp-ports 8000
```
Screenshot 2020-01-22 at 09 36 56

## Checklist:

I have:

- [ ] updated the documentation and/or roadmap (if required)
- [x] read the [CONTRIBUTION](https://github.com/inlets/inlets/blob/master/CONTRIBUTING.md) guide
- [x] signed-off my commits with `git commit -s`
- [ ] added unit tests

### CODE REVIEW

File: pkg/provision/ec2.go [hunk 1]
```diff
@@ -33,7 +33,11 @@ func (p *EC2Provisioner) Provision(host BasicHost) (*ProvisionedHost, error) {
 		return nil, err
 	}
 
-	groupID, name, err := p.securityGroup(host.Additional["inlets-port"])
+	port, err := strconv.Atoi(host.Additional["inlets-port"])
+	if err != nil {
+		return nil, err
+	}
+	groupID, name, err := p.creteEC2SecurityGroup(port)
```
Comment (alexellis) [inline pkg/provision/ec2.go]: crete or create?

Comment (adamjohnson01) [reply pkg/provision/ec2.go]: Thanks! I will update.

**Comment by adamjohnson01** [comment]
> Spotted a typo. What about the deletion?

Deletion happens in the Delete function

```
	for _, group := range groups {
		_, err := p.ec2Provisioner.DeleteSecurityGroup(&ec2.DeleteSecurityGroupInput{
			GroupId: group.GroupId,
		})
		if err != nil {
			return err
		}
	}
```

**Comment by adamjohnson01** [comment]
@alexellis , this needs to be merged in order for the operator to work correctly.

File: pkg/provision/ec2.go [hunk 2]
```diff
@@ -33,7 +33,11 @@ func (p *EC2Provisioner) Provision(host BasicHost) (*ProvisionedHost, error) {
 		return nil, err
 	}
 
-	groupID, name, err := p.securityGroup(host.Additional["inlets-port"])
+	port, err := strconv.Atoi(host.Additional["inlets-port"])
```
Comment (alexellis) [inline pkg/provision/ec2.go]: This might vary with inlets-pro, the port range is decided by the client

Comment (alexellis) [reply pkg/provision/ec2.go]: So for pro, we may want to open all ports.

Comment (adamjohnson01) [reply pkg/provision/ec2.go]: OK, I will update.

Comment (adamjohnson01) [reply pkg/provision/ec2.go]: I have added all ports if running pro.

**Comment by alexellis** [comment]
Thank you Adam. I can see the build is still pending. Ping me if it passes ok.

**Comment by adamjohnson01** [comment]
@alexellis , build passed!

**Comment by alexellis** [comment]
@adamjohnson01 @digitalsanctum @utsavanand2 I'm looking for docs on what kind of IAM role is required for the EC2 provisioner. Do either of you have a set of permissions we can add to the README?

**Pull request merged**